dotnet/src/Connectors/Connectors.Memory.Weaviate/WeaviateMemoryStore.cs

```diff
@@ -386,7 +386,29 @@ public async Task RemoveAsync(string collectionName, string key, CancellationTok
     /// <inheritdoc />
     public async Task RemoveBatchAsync(string collectionName, IEnumerable<string> keys, CancellationToken cancellationToken = default)
     {
-        await Task.WhenAll(keys.Select(async k => await this.RemoveAsync(collectionName, k, cancellationToken).ConfigureAwait(false))).ConfigureAwait(false);
+        Verify.NotNullOrWhiteSpace(collectionName, "Collection name is empty");
+        Verify.NotNull(keys, "Key is NULL");
+        var keyArray = keys.ToArray();
+        if (keyArray.Length == 0)
+        {
+            return;
```

Comment:
Return a completed task?

Reply:
FYI, returning a completed task will only work if we make the method non-async, but we have awaits elsewhere, so this probably won't be possible.

Reply:
@westey-m instead return can we do something like execute request if keyArray.Length>0 ?